block-storage/src/main/scala/coop/rchain/blockstorage/BlockMetadata.scala

```diff
@@ -1,19 +1,54 @@
 package coop.rchain.blockstorage
 
+import java.nio.ByteBuffer
+
 import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.BlockDagRepresentation.Validator
 import coop.rchain.blockstorage.BlockStore.BlockHash
 import coop.rchain.blockstorage.util.BlockMessageUtil
+import coop.rchain.blockstorage.util.byteOps._
 import coop.rchain.casper.protocol.{BlockMessage, Justification}
 
 final case class BlockMetadata(
     blockHash: BlockHash,
     parents: List[BlockHash],
     sender: ByteString,
     justifications: List[Justification],
-    weightMap: Map[ByteString, Long],
+    weightMap: Map[Validator, Long],
     blockNum: Long,
     seqNum: Int
-)
+) {
+  def toByteString: ByteString = {
```

Comment:
I wonder if this method, along with `fromBytes` would be an argument for just making this class be part of the protobuf model (then we would get them for free).

Reply:
I think this makes sense. It will also allow us to modify `BlockMetadata` structure seamlessly in the future.